transformer/diff.go

```diff
@@ -2,76 +2,112 @@ package transformer
 
 import (
 	"fmt"
+	"regexp"
+	"strconv"
 	"strings"
 )
 
 type DiffTransformer struct {
-	rawDiff   string
```

Comment:
Clarification: Is there a reason why 'fileDiffs' and 'segments' are being removed?

Reply:
I converted them to maps so they could save more data.